### PR TITLE
Python. Homework #2. Task #1 Порсев Егор

### PR DESCRIPTION
**Pull request by Furetur**

### CODE REVIEW

File: hw2/curry.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    :return: True if func accepts varargs else False
+    """
+    _, vararg, *rest = inspect.getfullargspec(func)
+    return vararg is not None
+
+
+R = TypeVar("R")
+
+
+def curry_internal(func: Callable[..., R], arity: int, passed_args: list = None) -> Callable:
```
Comment (nbirillo) [inline hw2/curry.py]: You can use `_` or `__` to mark it as an internal function

File: hw2/curry.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    def curried_function(x):
+        if len(passed_args) + 1 < arity:
+            return curry_internal(func, arity, [*passed_args, x])
+        else:
+            return func(*passed_args, x)
+
+    return curried_function
+
+
+def curry_explicit(func: Callable[..., R], arity: int) -> Callable:
```
Comment (nbirillo) [inline hw2/curry.py:68]: Please, try to fix your functions for this case:
```
f4 = curry_explicit(print, 4)
g4 = uncurry_explicit(f4, 4)
f2 = f4(10)(20)
g2 = uncurry_explicit(f2, 2)
print('---')
f4(5)(15)(25)(35)
f2(30)(40)
print('---')
g4(5, 15, 25, 35)
g2(30, 40)
```
The expected output in this case:
```
---
5 15 25 35
10 20 30 40
---
5 15 25 35
10 20 30 40
```
But in your case, you will have an error

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great work, but I found one wrong case

File: hw2/curry.py (hunk 2 above)
Comment (Furetur) [reply hw2/curry.py:68]: For some reason python raises an exception when I try to inspect the signature of the `print` function

Comment (Furetur) [reply hw2/curry.py:68]: The only way of fixing this error that I see is removing the arity-checking functionality completely.

Comment (Furetur) [reply hw2/curry.py:68]: No. I found a way of fixing it! Now `curry_explicit` ignores these kinds of errors when it cannot access signatures of builtin functions

**Review by nbirillo — APPROVED** [review]
Great! Thank you!

**Pull request merged**